src/packet/ipv4.rs

```diff
@@ -76,11 +79,28 @@ pub struct Ipv4Option {
     copied: u1,
     class: u2,
     number: u5,
-    length: u8,
+    #[length_fn = "ipv4_options_length_length"]
+    length: Vec<u8>,
+    #[length_fn = "ipv4_options_payload_length"]
     #[payload]
     data: Vec<u8>,
 }
 
+fn ipv4_options_length_length(option: &Ipv4OptionPacket) -> usize {
+    match option.get_number() {
+        0 => 0,
+        1 => 0,
+        _ => 1,
+    }
+}
+
+fn ipv4_options_payload_length(ipv4_options: &Ipv4OptionPacket) -> usize {
```

Comment:
Could you rename this to `ipv4_option_payload_length` (notice the lack of `s` after option).